src/content/docs/en/reference/programmatic-reference.mdx

```diff
@@ -0,0 +1,197 @@
+
+### `configFile`
+
+<p>
+
+**Type:** `string | false`<br />
+**Default:** `undefined`
+</p>
+
+A custom path to the Astro config file.
+
+If this value is undefined (default) or unset, Astro will search for an `astro.config.(js,mjs,ts,mts)` file relative to the `root` and load the config file if found.
+
+If a relative path is set, it will resolve based on the `root` option.
+
+Set to `false` to disable loading any config files.
+
+The inline config passed in this object will take highest priority when merging with the loaded user config.
+
+### `mode`
+
+<p>
+
+**Type:** `string`<br />
+**Default:** `"development"` when running `astro dev`, `"production"` when running `astro build`<br />
+<Since v="5.0.0" />
+</p>
+
+The mode used when developing or building your site (e.g. `"production"`, `"testing"`).
+
+This value is passed to Vite using [the `--mode` flag](#--mode-string) when the `astro build` or `astro dev` commands are run to determine the value of `import.meta.env.MODE`. This also determines which `.env` files are loaded, and therefore the values of `astro:env`. See the [environment variables page](/en/guides/environment-variables/) for more details.
+
+To output a development-based build, you can run `astro build` with the [`--devOutput` flag](#--devoutput).
+
+### `logLevel`
+
+<p>
+
+**Type:** `"debug" | "info" | "warn" | "error" | "silent"`<br />
+**Default:** `"info"`
+</p>
+
+The logging level to filter messages logged by Astro.
+
+- `"debug"`: Log everything, including noisy debugging diagnostics.
+- `"info"`: Log informational messages, warnings, and errors.
+- `"warn"`: Log warnings and errors.
+- `"error"`: Log errors only.
+- `"silent"`: No logging.
+
+## `dev()`
+
+**Type:** `(inlineConfig: AstroInlineConfig) => Promise<DevServer>`
+
+Similar to [`astro dev`](#astro-dev), it runs Astro's development server.
+
+```js
+import { dev } from "astro";
+
+const devServer = await dev({
+  root: "./my-project",
+});
+
+// Stop the server if needed
+await devServer.stop();
+```
+
+### `DevServer`
+
+```ts
+export interface DevServer {
+	address: AddressInfo;
+	handle: (req: http.IncomingMessage, res: http.ServerResponse<http.IncomingMessage>) => void;
+	watcher: vite.FSWatcher;
+	stop(): Promise<void>;
+}
+```
+
+#### `address`
+
+The address the dev server is listening on.
+
+This property contains the value returned by Node's [`net.Server#address()` method](https://nodejs.org/api/net.html#serveraddress).
+
+#### `handle()`
+
+A handle for raw Node HTTP requests. You can call `handle()` with an [`http.IncomingMessage`](https://nodejs.org/api/http.html#class-httpincomingmessage) and an [`http.ServerResponse`](https://nodejs.org/api/http.html#class-httpserverresponse) instead of sending a request through the network.
+
+#### `watcher`
+
+The [Chokidar file watcher](https://github.com/paulmillr/chokidar#getting-started) as exposed by [Vite's development server](https://vite.dev/guide/api-javascript#vitedevserver).
+
+#### `stop()`
```

Comment:
I have a doubt here. Usually for functions we use the following format: `() => Promise<void>` but the interface above does not match. I checked in [`withastro/astro` repo](https://github.com/withastro/astro/blob/25baa4ed0c5f55fa85c2c7e2c15848937ed1dc9b/packages/astro/src/core/dev/dev.ts#L24) and... TIL, `stop(): Promise<void>` seems valid. So I don't know if we should use:
* `**Type:** () => Promise<void>`
* or `**Type:** Promise<void>`

```suggestion
#### `stop()`

<p>

**Type:** `Promise<void>`
</p>
```

Reply:
Alright, then I say we use this until someone tells us it's wrong! 😅